### DISCUSSION TITLE
Button types seem too opinionated

### DISCUSSION THREAD

**Discussion by bzupancic** [IgniteUI/igniteui-angular#11172] [Ideas]
I know that Material uses similar naming for their button types, but when it comes to a highly customizable control library this naming convention is too opinionated.

// example
If I want a simple button with a background and no elevation, I am forced to write custom CSS (which is fine), and use a naming convention that doesn't match the style.

// suggestion
Either add a 'Contained' button type (like Material), or better yet move to a more semantic naming convention where the control is back in the users hand i.e. <igx-button kind="outlined" variant="secondary"></button>

**Comment by simeonoff** [comment]
Hi @bzupancic, thanks for the feedback. Removing the existing types will be a big breaking change, we can however replace `raised` with `contained` so that it is more in-line with the latest material naming convention.
